### DISCUSSION TITLE
You rush a Miracle Man, you get rotten miracles

### DISCUSSION THREAD

**Discussion by joshbruce** [markedjs/marked#1643] [Show and tell]
![miracles](https://user-images.githubusercontent.com/15252830/79144505-b12ea280-7d84-11ea-9c4c-18cd7cb060d2.gif)

Please see previous #1522 for backstory.

December 1st, 2017 the core team came together to take on Marked. Since then a lot has happened...a lot. Some changes in direction we wanted to talk about:

1. Spec-compliance, semver, and a 1.0 release
2. Make Marked more extensible rather than keeping up with the world

## Spec-compliance, semver, and a 1.0 release

Initially the community and core team wanted to ensure spec-compliance with CommonMark and GFM before pushing a 1.0 release. This compliance would be a community-driven effort. As we have published new releases we have received feedback regarding concerns about breaking changes and semantic versioning despite being in compliance with the [zero-major definition](https://semver.org/#spec-item-4).

Given many of Marked's dependents are using Marked in production and have been for years, we want to flip-the-script (see what I did there?) and do a 1.0 release shortly: Letting Marked be Marked.

This doesn't mean Marked becomes static; instead, the community and core team can focus on improving developer experience, extensibility, and making Marked easier to customize.

We will not decline PRs improving spec-compliance, it just won't necessarily be the push.

If you require spec-compliance for implementation, please see alternative libraries: [CommonMakeJS](https://github.com/commonmark/commonmark.js/) and [Markdown-IT](https://github.com/markdown-it/markdown-it) come recommended by the community and core team.

**The core team would like to open a discussion (albeit brief) on the breakpoint for releasing the 1.0 version of Marked to usher in improved communication through version numbers.**

## Make Marked more extensible and modular

See the beginning the community has discussed making Marked more modular and extensible. This should bring a fair balance between the speed afforded by Marked while allowing increased customization. For example, it may be possible for developers using Marked to use the Lexer from Marked while plugging in their own parser.

We would still want to maintain the three key things that seem to make Marked, Marked:

1. Few dev-dependencies
2. Lightweight
3. Fast

## Further considerations

All comments, as always, welcome and I did not run this post by the rest of the core team before posting just basing it off the brief conversation we had elsewhere.

We'll be updating some of the milestone references (possibly removing them for now) and continuing to make MarkedJS a better product, community, and project with the understanding this is a labor of love for all involved and different aspects of the project.

Big thanks to the community and core team for the couple of years!

**Comment by Pratik-ahirrao** [comment]
Can i work on this issue?

**Comment by UziTech** [comment]
@Pratik-ahirrao the best way to work on this is to find an issue and submit a PR that fixes it.

If you want to help make marked spec compliant you can change the specs that have `shouldFail: true` to `shouldFail: false` and try to make the tests pass.

https://github.com/markedjs/marked/blob/7b3036f8c0440cfd003ce47dd6e1a92af0f5e822/test/specs/commonmark/commonmark.0.29.json#L404-L412

**Comment by Pratik-ahirrao** [comment]
I have changed the specs that had shouldFail: true to shouldFail: false.
Now what should i do?

**Comment by Pratik-ahirrao** [comment]
Shall i make a PR?

**Comment by joshbruce** [comment]
Yes, please do! The core team will take a look and work to resolution with you.

**Comment by Pratik-ahirrao** [comment]
I have send the pull request

**Comment by Pratik-ahirrao** [comment]
I have done what you said but i think PR has failed. What should i do now?

**Comment by Pratik-ahirrao** [comment]
test cases are failed.

**Comment by Pratik-ahirrao** [comment]
I am getting problem while running the script in my laptop.

**Comment by Pratik-ahirrao** [comment]
I am making a new PR. Please check and please tell me that its working.
